nautobot/ipam/models.py

```diff
@@ -960,6 +961,20 @@ def get_utilization(self):
         return UtilizationData(numerator=numerator_set.size, denominator=denominator)
 
 
+@extras_features("graphql")
+class PrefixLocationAssignment(BaseModel):
+    # Using `related_name="+"`` because prefix is the parent of this through model and can only be accessed by Prefix from its `locations` field.
+    # i.e Prefix.objects.filter(locations__in=...), Prefix.objects.first().locations etc.
+    prefix = models.ForeignKey("ipam.Prefix", on_delete=models.CASCADE, related_name="+")
+    location = models.ForeignKey("dcim.Location", on_delete=models.CASCADE, related_name="prefix_assignments")
+
+    class Meta:
+        unique_together = ["prefix", "location"]
```

Comment:
Let's be sure to define an `ordering` on these models, that may fix the unittest issue as well.